warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IngestTypeVisitor.java

```diff
@@ -385,7 +387,9 @@ private Set<String> getIngestTypesForIntersection(ASTAndNode node) {
 
     private Set<String> intersectTypes(Set<String> typesA, Set<String> typesB) {
         if (typesA.contains(UNKNOWN_TYPE) || typesB.contains(UNKNOWN_TYPE)) {
-            return Collections.singleton(UNKNOWN_TYPE);
+            Set<String> unknown = new HashSet<>();
```

Comment:
Could you explain the benefit of going from:
```
            return Collections.singleton(UNKNOWN_TYPE);
``` 
to 
```
Set<String> unknown = new HashSet<>();
unknown.add(UNKNOWN_TYPE);
return unknown;
```
Is there a reason to go from an immutable Set<String> to a mutable Set<String> if all it's doing is returning one with the UNKNOWN_TYPE?

Reply:
The result will likely have set operations performed (add, addAll, retain) so it can't be immutable.